project/build.scala

```diff
@@ -1,6 +1,147 @@
-package object build extends ScalamacrosBuild
+object ScalamacrosBuildPlugin extends AutoPlugin {
+  import Version._
+  override def requires: Plugins = JvmPlugin && BuildInfoPlugin
+  override def trigger: PluginTrigger = allRequirements
+
+  object autoImport {
+    lazy val pluginSettings = Def.settings(
+      crossVersion := CrossVersion.full,
+      unmanagedSourceDirectories.in(Compile) += {
+        val base = sourceDirectory.in(Compile).value
+        base / ("scala-" + scalaVersion.value)
+      },
+      test.in(assembly) := {},
+      logLevel.in(assembly) := Level.Error,
+      assemblyJarName.in(assembly) := {
+        name.value + "_" + scalaVersion.value + "-" + version.value + "-assembly.jar"
+      },
+      assemblyOption.in(assembly) ~= { _.copy(includeScala = false) },
+      Keys.`package`.in(Compile) := {
+        val slimJar = Keys.`package`.in(Compile).value
+        val fatJar = new File(crossTarget.value + "/" + assemblyJarName.in(assembly).value)
+        val _ = assembly.value
+        IO.copy(List(fatJar -> slimJar), overwrite = true)
+        slimJar
+      },
+      packagedArtifact.in(Compile).in(packageBin) := {
+        val temp = packagedArtifact.in(Compile).in(packageBin).value
+        val (art, slimJar) = temp
+        val fatJar = new File(crossTarget.value + "/" + assemblyJarName.in(assembly).value)
+        val _ = assembly.value
+        IO.copy(List(fatJar -> slimJar), overwrite = true)
+        (art, slimJar)
+      }
+    )
+
+    lazy val noPublish = Def.settings(
+      publishArtifact.in(Compile, packageDoc) := false,
+      publishArtifact.in(packageDoc) := false,
+      sources.in(Compile, doc) := Seq.empty,
+      packagedArtifacts := Map.empty,
+      publishArtifact := false,
+      publish := {}
+    )
+  }
+
+  override def globalSettings: Seq[Def.Setting[_]] = List(
+    scalaVersion := scala212,
+    crossScalaVersions := List(scala212),
+    crossVersion := CrossVersion.binary,
+    scalacOptions ++= Seq("-deprecation", "-feature", "-unchecked", "-Xfatal-warnings"),
+    logBuffered := false,
+    incOptions := incOptions.value.withLogRecompileOnMacro(false),
+    updateOptions := updateOptions.value.withCachedResolution(true),
+    triggeredMessage.in(ThisBuild) := Watched.clearWhenTriggered,
+    libraryDependencies ++= List(
+      "junit" % "junit" % "4.12" % Test,
+      "com.novocode" % "junit-interface" % "0.11" % Test
+    ),
+    testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-n"),
+    organization := "org.scalamacros",
+    licenses := Seq(
+      "BSD" -> url("https://github.com/scalamacros/scalamacros/blob/master/LICENSE.md")
+    ),
+    homepage := Some(url(s"https://github.com/scalacenter/macros")),
+    scmInfo := Some(
+      ScmInfo(
+        url(s"https://github.com/scalacenter/macros"),
+        "scm:git:git@github.com:scalacenter/macros.git"
+      )
+    ),
+    bintrayOrganization := Some("scalamacros"),
+    publishArtifact.in(Compile) := true,
```

Comment:
I think these two do not do anything here.

Reply:
Removed. Copied from old build.